### PR TITLE
transition to testthat 3e

### PR DESCRIPTION
**Pull request by simonpcouch**
Mostly just a version bump and dropping `context`! Tests made good use of `expect_is`, and did so in a few different ways, so replaced with `expect_s3_class`, `expect_equal`, and `expect_type` depending on the situation.🌼

### CODE REVIEW

**Comment by simonpcouch** [comment]
Some build failures addressed in #106, some related to:

```
   x Cannot install packages:
  * local::.: Can't install dependency randomForest
  * randomForest: Needs R >= 4.1.0 
```

**Comment by juliasilge** [comment]
You might check out how you can handle randomForest not being available on older R, like [in hardhat](https://github.com/tidymodels/butcher/blob/main/.github/workflows/R-CMD-check.yaml).

**Comment by simonpcouch** [comment]
Ah, I see! Thank you for passing this along—will mark #106 as draft and make those changes there.

**Review by juliasilge — CHANGES_REQUESTED** [review]
This is looking good! 🙌 

Part of what we need to get fixed in the very short term (see #104) are the tests for the generated SQL. Can you also find all the tests that are for generated SQL / dbplyr / etc and change those to snapshot tests?

**Comment by simonpcouch** [comment]
Sure thing! Made that happen in [620e316](https://github.com/tidymodels/tidypredict/pull/105/commits/620e316f252a46792337cceeb76f861c68a3f84d)—now closes #104.

**Comment by simonpcouch** [comment]
Ah, yes! Checking out those and other `expect_silent`s, these don’t seem particularly vulnerable to dependency breakage in a meaningful way and are worth checking more thoroughly. Converted those to `expect_snapshot`. :)

File: tests/testthat/_snaps/earth.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Model can be saved and re-loaded
+
+    Code
+      tidypredict_fit(pm)
+    Output
+      list()
```
Comment (juliasilge) [inline tests/testthat/_snaps/earth.md:6]: This one seems like it does not actually work? Can you double check that this is a bug? I [confirm that `tidypredict_fit(model)` and `tidypredict_sql(model, dbplyr::simulate_odbc())` both work](https://tidypredict.tidymodels.org/articles/mars.html), but it seems like `model %>% parse_model() %>% tidypredict_fit()` does not work for earth?

If you can confirm this, then let's open an issue about fixing this problem in another PR later.

**Review by juliasilge — APPROVED** [review]

**Comment by simonpcouch** [comment]
Ah, and some of these snapshots ended up failing on Actions. Looks like some line wrapping issues, but some actual differences in model results (maybe RNG?).

File: tests/testthat/_snaps/earth.md (hunk 1 above)
Comment (simonpcouch) [reply tests/testthat/_snaps/earth.md:6]: Confirming that I see this locally as well.🤔

**Comment by simonpcouch** [comment]
Okay, doesn't look like the failing snapshot tests are RNG-, rounding error-, or line wrapping-related.

I'd be glad to revert the failing tests and file issues if that seems like a good route forward! Will defer to others with more expertise on this package to debug those snapshot differences (if they indeed are issues at all).

**Comment by simonpcouch** [comment]
Okay, just reverted a few of those tests!

Thumbs up from me on merging if this looks good to you—filing an issue re: the snapshot failures we saw in a moment.

**Review by juliasilge — APPROVED** [review]
Looking good! 🚀

**Pull request merged**